### PR TITLE
add implementation for webp

### PR DESCRIPTION
**Pull request by chris-kruining**
#77 Made a pull request, I believe this is a complete implementation, however a code review would be appreciated

### CODE REVIEW

File: Image.php [hunk 1]
```diff
@@ -645,7 +654,7 @@ public function save($file, $type = 'guess', $quality = 80)
         if ($type == 'guess') {
             $type = $this->guessType();
         }
-
+
```
Comment (Soullivaneuh) [inline Image.php]: Trailing space here.

File: Adapter/Imagick.php [hunk 2]
```diff
@@ -392,6 +392,11 @@ protected function openPng($file)
         // TODO: Implement openPng() method.
     }
 
+    protected function openWebp($file)
+    {
+        // TODO: Implement openWebp() method.
```
Comment (Soullivaneuh) [inline Adapter/Imagick.php]: Can't this be handle now?

File: Image.php [hunk 3]
```diff
@@ -674,6 +683,10 @@ public function save($file, $type = 'guess', $quality = 80)
                 $success = $this->getAdapter()->savePng($file);
             }
 
+            if ($type == 'webp') {
```
Comment (Soullivaneuh) [inline Image.php]: Please use strict comparison.

File: Adapter/Imagick.php (hunk 2 above)
Comment (chris-kruining) [reply Adapter/Imagick.php]: I didn't wat to stray from the rest of the code base so I did not implement this

File: Image.php (hunk 3 above)
Comment (chris-kruining) [reply Image.php]: Would love to, but the rest of the checks don't either, and I went for code consistency, imho this is for another PR, one that cleans up the code base

Comment (garygreen) [reply Image.php]: @Soullivaneuh honestly. I think your being overly critial in these reviews considering the codebase for Gregwar doesn't follow strict standards and it's barely even maintained by it's owner anymore. I would be thankful for any collaboration from maintainers.

Comment (Gregwar) [reply Image.php]: @Soullivaneuh I am ok to merge and fix this in another PR, along with code standard if needed

**Comment by chris-kruining** [comment]
ping

**Comment by Gregwar** [comment]
Thanks for contributing, @soullivaneuh  and @garygreen, let's open other issues if you want this lib to comply with coding standards as mentionned

**Comment by aweelex** [comment]
@Gregwar Any tag for web support?

**Comment by Gregwar** [comment]
Tagged `v2.0.25`

**Pull request merged**